hibernate-reactive-core/src/test/java/org/hibernate/reactive/ORMReactivePersistenceTest.java

```diff
@@ -0,0 +1,187 @@
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import io.vertx.ext.unit.TestContext;
+
+import static org.hibernate.reactive.containers.DatabaseConfiguration.DBType.POSTGRESQL;
+import static org.hibernate.reactive.util.impl.CompletionStages.completedFuture;
+
+public class ORMReactivePersistenceTest extends BaseReactiveTest {
+
+	@Rule
+	public DatabaseSelectionRule rule = DatabaseSelectionRule.runOnlyFor( POSTGRESQL );
+
+	private SessionFactory ormFactory;
+
+	@Override
+	protected Configuration constructConfiguration() {
+		Configuration configuration = super.constructConfiguration();
+		configuration.addAnnotatedClass( Flour.class );
+		return configuration;
+	}
+
+	@Before
+	public void prepareOrmFactory() {
+		Configuration configuration = constructConfiguration();
+		configuration.setProperty( Settings.DRIVER, "org.postgresql.Driver" );
+		configuration.setProperty( Settings.DIALECT, "org.hibernate.dialect.PostgreSQL95Dialect");
+
+		StandardServiceRegistryBuilder builder = new StandardServiceRegistryBuilder()
+				.applySettings( configuration.getProperties() );
+
+		StandardServiceRegistry registry = builder.build();
+		ormFactory = configuration.buildSessionFactory( registry );
+	}
+
+	@After
+	public void cleanDb(TestContext context) {
+		ormFactory.close();
+		test( context, completedFuture( openSession() )
+				.thenCompose( s -> s.createQuery( "delete Flour" )
+						.executeUpdate() ) );
+
+	}
+
+	@Test
+	public void testORMWithStageSession(TestContext context) {
+		final Flour almond = new Flour( 1, "Almond", "made from ground almonds.", "Gluten free" );
+
+		Session session = ormFactory.openSession();
+		session.beginTransaction();
+		session.persist( almond );
+		session.getTransaction().commit();
+		session.close();
+
+		// Check database with Stage session and verify 'almond' flour exists
+		Stage.Session stageSession = openSession();
+		test( context, stageSession.find( Flour.class, almond.id )
+				.thenAccept( entityFound -> context.assertEquals( almond, entityFound ) )
+		);
+	}
+
+	@Test
+	public void testORMWitMutinySession(TestContext context) {
+		final Flour rose = new Flour( 2, "Rose", "made from ground rose pedals.", "Full fragrance" );
+		final Flour sunflower = new Flour( 3, "Sunflower", "made from ground sunflowers.", "Seedy" );
+
+		Session ormSession = ormFactory.openSession();
+		ormSession.beginTransaction();
+		ormSession.persist( rose );
+		ormSession.getTransaction().commit();
+		ormSession.close();
+
+		// Check database with Mutiny session and verify 'rose' flour exists
+		Mutiny.Session mutinySession = openMutinySession();
+		// find the entity using the id
+		test( context, mutinySession.find( Flour.class, rose.id)
+				// and check it's the right one
+				.invoke( foundRose -> context.assertEquals( rose, foundRose ) )
+				// and that it's not the wrong one
+				.invoke( noAlmond -> context.assertNotEquals(rose, sunflower))
```

Comment:
Is it really useful to test that the two objects are different? You have already checked that rose is the expected one.
The only way this assert can fail is if there is something wrong in the test.
It doesn't seem very helpful.